plugins/filter/listify.py

```diff
@@ -0,0 +1,152 @@
+        type: raw
+        required: True
+      keys:
+        description: Comma separated keys of type string denoting the ACI objects.
+        required: True
+"""
+
+EXAMPLES = r"""
+- name: Set vars
+  ansible.builtin.set_fact:
+    data:
+      tenant:
+      - name: ansible_test
+        description: Created using listify
+        app:
+        - name: app_test
+          epg:
+          - name: web
+            bd: web_bd
+          - name: app
+            bd: app_bd
+        bd:
+        - name: bd_test
+          subnet:
+          - name: 10.10.10.1
+            mask: 24
+            scope: private
+          vrf: vrf_test
+        - name: bd_test2
+          subnet:
+          - name: 20.20.20.1
+            mask: 24
+            scope: public
+          vrf: vrf_test
+        vrf:
+        - name: vrf_test
+
+- name: Create tenants
+  cisco.aci.aci_tenant:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: '{{ item.tenant_name }}'
+    description: '{{ item.tenant_description }}'
+  with_items: '{{ data|cisco.aci.aci_listify("tenant") }}'
+
+- name: Create VRFs
+  cisco.aci.aci_vrf:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: '{{ item.tenant_name }}'
+    vrf_name: '{{ item.tenant_vrf_name }}'
+  with_items: '{{ data|cisco.aci.aci_listify("tenant","vrf") }}'
+
+- name: Create BDs
+  cisco.aci.aci_bd:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: '{{ item.tenant_name }}'
+    vrf: '{{ item.tenant_bd_vrf }}'
+    bd: '{{ item.tenant_bd_name }}'
+    enable_routing: yes
+  with_items: '{{ data|cisco.aci.aci_listify("tenant","bd") }}'
+
+- name: Create BD subnets
+  cisco.aci.aci_bd_subnet:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: '{{ item.tenant_name }}'
+    bd: '{{ item.tenant_bd_name }}'
+    gateway: '{{ item.tenant_bd_subnet_name }}'
+    mask: '{{ item.tenant_bd_subnet_mask }}'
+    scope: '{{ item.tenant_bd_subnet_scope }}'
+  with_items: '{{ data|cisco.aci.aci_listify("tenant","bd","subnet") }}'
+
+- name: Create APs
+  cisco.aci.aci_ap:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: '{{ item.tenant_name }}'
+    app_profile: '{{ item.tenant_app_name }}'
+  with_items: '{{ data|cisco.aci.aci_listify("tenant","app") }}'
+
+- name: Create EPGs
+  cisco.aci.aci_epg:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: '{{ item.tenant_name }}'
+    app_profile: '{{ item.tenant_app_name }}'
+    epg: '{{ item.tenant_app_epg_name }}'
+    bd: '{{ item.tenant_app_epg_bd }}'
+  with_items: '{{ data|cisco.aci.aci_listify("tenant","app","epg") }}'
+"""
+
+
+def listify(d, *keys):
+    return listify_worker(d, keys, 0, [], {}, "")
+
+
+def listify_worker(d, keys, depth, result, cache, prefix):
```

Comment:
Have you thought of using a [`generator iterator`](https://docs.python.org/3/glossary.html#term-generator-iterator) for this? 
I have seen some examples of recursive functions using the generator pattern. The benefit would be less memory usage and potentially better performance because you may not need to worry about the cache or keeping a results list.

Reply:
That makes sense. I'll modify the code and test it out.